spec_insert/Rakefile

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+require 'rake'
+
+desc 'Download the OpenSearch API specification'
+task :download_spec do
+  sh 'curl -L -X GET ' \
+     'https://github.com/opensearch-project/opensearch-api-specification' \
+     '/releases/download/main-latest/opensearch-openapi.yaml ' \
+     '-o opensearch-openapi.yaml'
+end
```

Comment:
Download to a temp or `_site` folder to avoid `git add-ing` it later.

Don't use `sh curl`, it will be harder to debug, may not work on Windows, etc.

```ruby
openapi_url = "https://github.com/opensearch-project/opensearch-api-specification/releases/download/main-latest/opensearch-openapi.yaml"
openapi_path = "_site/opensearch-openapi.yaml"
File.write(openapi_path, URI.open(openapi_url).read)
```